casper/protocols/sharding/sharding_plot_tool.py

```diff
@@ -0,0 +1,105 @@
+"""The blockchain plot tool implements functions for plotting sharded blockchain data structures"""
+
+from casper.plot_tool import PlotTool
+from casper.safety_oracles.clique_oracle import CliqueOracle
+import casper.utils as utils
+
+
+class ShardingPlotTool(PlotTool):
+    """The module contains functions for plotting a blockchain data structure"""
+
+    def __init__(self, display, save, view, validator_set):
+        super().__init__(display, save, 's')
+        self.view = view
+        self.validator_set = validator_set
+        self.starting_blocks = self.view.starting_blocks
+        self.message_fault_tolerance = dict()
+
+        self.blockchain = []
+        self.communications = []
+
+        self.block_fault_tolerance = {}
+        self.message_labels = {}
+        self.justifications = {
+            validator: []
+            for validator in validator_set
+        }
+
+    def update(self, new_messages=None):
+        """Updates displayable items with new messages and paths"""
+        return
+
+        if new_messages is None:
+            new_messages = []
+
+        self._update_new_justifications(new_messages)
+        self._update_blockchain(new_messages)
+        self._update_block_fault_tolerance()
+        self._update_message_labels(new_messages)
+
+    def plot(self):
+        """Builds relevant edges to display and creates next viewgraph using them"""
+        return
```

Comment:
Typo perhaps? Or do you want to return immediately

Reply:
Currently don't have graph plotting on the sharding consensus, just some terminal printing. So this is a stub